### DISCUSSION TITLE
API Management Self Hosted Gateway Send One Way Request Response Timeout

### DISCUSSION THREAD

**Discussion by lvoneye** [Azure/api-management-self-hosted-gateway#254] [Q&A]
We are seeing abnormal behavior when using either the send-request or send-one-way-request policy as part of an outbound section of an APIM hosted API with the self-hosted gateway in our on-prem installation.   This behavior is known to only be happening to a single hosted API at this time, and even then only to a subset of methods on that API.  It also happens randomly and intermittently, at a less than 1% of responses.  However, we are pushing 100,000s of thousands of requests daily so that presents itself negatively in our environment.

Specifically, the self-hosted gateway accepts the request, forwards it to the backend, which fulfills the request in under a second, and then the self-hosted gateway appears to "hold" the response for 300 seconds (default timeout) before actually returning the response to the client.  Now because most of our clients do not have a timeout anywhere near 300 seconds, they are erroring and closing the client connection well before that 300 second elapsed time.  We were able to prove that APIM would return a response after 300 seconds by writing a PowerShell script that calls an API hosted in the gateway repeatedly, with a timeout longer than 300 seconds, until it holds that response for the 300 seconds.

There is a release note in "Container-v2.2.0" that says "Fixed a bug where send-one-way-request policy does not gracefully sending a response due to an empty stack" which we appear to still be experiencing similar behavior, potentially, in 2.2 and 2.3.  Is it possible that that condition could still be triggered in these container versions?

**Comment by tomkerkhove** [comment]
Covered in #226
